adaptdl/adaptdl/speedup.py

```diff
@@ -135,13 +152,22 @@ def __call__(self, nodes, replicas, return_local_bsz=False):
         elif self._elastic_bsz:
             max_local_bsz = np.floor(self._max_batch_size / replicas)
             min_local_bsz = np.ceil(self._init_batch_size / replicas)
-            if self._max_local_bsz is not None:
+            if (self._max_local_bsz is not None and
+                    not self._accumulation):
                 max_local_bsz = np.minimum(self._max_local_bsz, max_local_bsz)
             if self._min_local_bsz is not None:
                 min_local_bsz = np.maximum(self._min_local_bsz, min_local_bsz)
             assert np.all(max_local_bsz >= min_local_bsz)
             # Sample a bunch of potential local_bsz values
             local_bsz = np.geomspace(min_local_bsz, max_local_bsz, num=100)
+            local_bsz = np.where(
+                replicas == 1,
+                np.append(
+                    np.broadcast_to(self._init_batch_size,
+                                    (1, max_local_bsz.size)),
+                    np.geomspace(min_local_bsz * 2, max_local_bsz, 99),
+                    axis=0),
+                local_bsz)
```

Comment:
Still confused about the logic here. When replicas==1, shouldn't we be looking for the best (atomic_bsz, accum_steps) where either (1) atomic_bsz==init_batch_size OR (2) accum_steps>=1? Not sure how the defined range of local_bsz can fully explore those pairs.